frictionless/resource.py

```diff
@@ -635,6 +635,24 @@ def from_storage(storage, *, name):
         """
         return storage.read_resource(name)
 
+    @staticmethod
+    def from_ckan(*, base_url, resource_id, api_key=None):
```

Comment:
* `dataset_id` is required for `Package.from_ckan`
* For `Package.to_ckan` and `Resource.to_ckan`, `dataset_id` is not required, but providing it allows us to reduce the search space (so we need to make fewer calls to `datastore_search` if we have it)
* For `Resource.from_ckan`, `dataset_id` would do nothing, so I haven't bothered allowing `dataset_id` as a param to this method